integration/metrics_test.go

```diff
@@ -0,0 +1,221 @@
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/eclipse-kanto/kanto/integration/util"
+	"github.com/eclipse-kanto/system-metrics/internal/metrics"
+
+	"github.com/eclipse/ditto-clients-golang/protocol"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	metricsFeatureID = "Metrics"
+	dataAction       = "data"
+)
+
+type systemMetricsSuite struct {
+	suite.Suite
+
+	util.SuiteInitializer
+
+	metricsThingURL   string
+	metricsFeatureURL string
+
+	pathData  string
+	topicData string
+}
+
+func (suite *systemMetricsSuite) SetupSuite() {
+	suite.SuiteInitializer.Setup(suite.T())
+
+	suite.metricsThingURL = util.GetThingURL(suite.Cfg.DigitalTwinAPIAddress, suite.ThingCfg.DeviceID)
+	suite.metricsFeatureURL = util.GetFeatureURL(suite.metricsThingURL, metricsFeatureID)
+
+	suite.pathData = util.GetFeatureOutboxMessagePath(metricsFeatureID, dataAction)
+	suite.topicData = util.GetLiveMessageTopic(suite.ThingCfg.DeviceID, protocol.TopicAction(dataAction))
+}
+
+func (suite *systemMetricsSuite) TearDownSuite() {
+	suite.SuiteInitializer.TearDown()
+}
+
+func TestSystemMetricsSuite(t *testing.T) {
+	suite.Run(t, new(systemMetricsSuite))
+}
+
+func (suite *systemMetricsSuite) testMetrics(params map[string]interface{}, originators ...string) error {
+	ws, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, "failed to create websocket connection")
+	defer ws.Close()
+
+	err = util.SubscribeForWSMessages(suite.Cfg, ws, "START-SEND-MESSAGES", "")
+	require.NoError(suite.T(), err, "unable to listen for events by using a websocket connection")
+
+	defer func() {
+		_, err := util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", map[string]interface{}{
+			"frequency": "0s",
+		})
+		assert.NoError(suite.T(), err, "error while stopping system metrics")
+	}()
+
+	_, err = util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", params)
+	require.NoError(suite.T(), err, "error while requesting the system metrics")
+
+	result := util.ProcessWSMessages(suite.Cfg, ws, func(msg *protocol.Envelope) (bool, error) {
+		if msg.Path != suite.pathData {
+			return false, nil
+		}
+
+		if msg.Topic.String() != suite.topicData {
+			return false, nil
+		}
+
+		data, err := json.Marshal(msg.Value)
+		if err != nil {
+			return true, err
+		}
+
+		metric := new(metrics.MetricData)
+		if err := json.Unmarshal(data, metric); err != nil {
+			return true, err
+		}
+
+		var originatorFound bool
+		for _, m := range metric.Snapshot {
+
+			for _, originator := range originators {
+				if originator == m.Originator {
+					originatorFound = true
+				}
+			}
+
```

Comment:
If the originator is not found after this loop, TC has to fail. originatorFound can be removed then.